lib/rubocop/cop/lint/overwrite_by_rescue.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks for overwriting an exception with an exception result by use `rescue =>`.
+      #
+      # @example
+      #
+      #   # bad
+      #   begin
+      #     something
+      #   rescue => StandardError
+      #   end
+      #
+      #   # good
+      #   begin
+      #     something
+      #   rescue StandardError
+      #   end
+      #
+      class OverwriteByRescue < Base
+        extend AutoCorrector
+        include RangeHelp
+
+        MSG = 'Override exception by `rescue =>`.'
```

Comment:
I guess the message can be more specific about what we actually overwrote here.

Reply:
A message is now show to indicate what is being overwritten by `rescue =>`.